mesonbuild/dependencies/misc.py

```diff
@@ -60,6 +60,25 @@ def netcdf_factory(env: 'Environment',
     return candidates
 
 
+class DlBuiltinDependency(BuiltinDependency):
+    def __init__(self, name: str, env: 'Environment', kwargs: T.Dict[str, T.Any]):
+        super().__init__(name, env, kwargs)
+
+        if self.clib_compiler.has_function('dlopen', '#include <dlfcn.h>', env)[0]:
```

Comment:
including dlfcn.h isn't strictly always necessary, although there is nothing wrong with doing so. There is a dummy/stub declaration that is used instead, if you don't include a header.

https://github.com/mesonbuild/meson/blob/5d932f31b985e46a4995d3c02b021cc94bef901d/mesonbuild/compilers/mixins/clike.py#L816-L823

Feel free to ignore this comment entirely. Arguably adding the include means you do a free check that the header is available (it should never be missing, but who knows).

Reply:
I'll think about it now, thank you.

Reply:
```
-        if self.clib_compiler.has_function('dlopen', '#include <dlfcn.h>', env)[0]:
+        if self.clib_compiler.has_function('dlopen', '', env)[0]:
```

With this patch, I had a successful mpv build on Archlinux. However, this is not a sign of success, because I have glibc 2.33, i.e. libdl is present in the system.
I can check on OpenBSD, which doesn't have this library, only later. I'll be sure to write here about the results as soon as I do it.

Intuitively it seems to me that the variant with an explicit specification of the title '#include <dlfcn.h>' is more correct, but I do not have a good reason for this position.
If an OpenBSD build with these modifications succeeds, I'm willing to add them to my patch.

Reply:
I think that if the OpenBSD build will succeed with this patch, there is no reason to do otherwise. Your advice is very valuable - thank you very much.
I'll be able to test the patch later tonight (MSK), then I'll post the final version.